contrib/src/templates/template_engine.rs

```diff
@@ -0,0 +1,55 @@
+use rocket::http::Status;
+use rocket::outcome;
+use rocket::request::FromRequest;
+use rocket::request::Outcome;
+use rocket::Request;
+use rocket::State;
+use templates::context::Context;
+
+/// The TemplateEngine type: implements `FromRequest`, allowing you to communicate
+/// directly with the template engines.
+///
+/// Using it in a request handler always returns successfully except when the [`Template::fairing()`]
+/// wasn't attached to rocket.
+pub struct TemplateEngine<'a> {
+    ctxt: State<'a, Context>,
+}
+
+impl<'a> TemplateEngine<'a> {
+    /// Checks whether a template exists with the given `name`.
+    ///
+    /// # Example
+    ///
+    /// ```rust,ignore
+    /// #[get("/")]
+    /// fn homepage(engine: TemplateEngine) -> Template {
+    ///     if engine.template_exists("specific") {
+    ///         return Template::render("specfic", json!({}));
+    ///     }
+    ///     Template::render("fallback", json!({}))
+    /// }
+    /// ```
+    pub fn template_exists(&self, name: &str) -> bool {
+        self.ctxt.templates.contains_key(name)
+    }
+
+    fn get_context(req: &'a Request) -> Result<State<'a, Context>, Status> {
```

Comment:
We have code similar to this one in template [mods.rs](https://github.com/SergioBenitez/Rocket/blob/master/contrib/lib/src/templates/mod.rs#L332). In fact that file contains it at 2 different locations. 

Maybe we want to create a global function to get the context in these 3 cases?